datafusion/core/src/physical_plan/sorts/sort_preserving_merge.rs

```diff
@@ -543,22 +559,22 @@ impl SortPreservingMergeStream {
             }
         }
 
-        loop {
-            // NB timer records time taken on drop, so there are no
-            // calls to `timer.done()` below.
-            let elapsed_compute = self.tracking_metrics.elapsed_compute().clone();
-            let _timer = elapsed_compute.timer();
+        // NB timer records time taken on drop, so there are no
+        // calls to `timer.done()` below.
+        let elapsed_compute = self.tracking_metrics.elapsed_compute().clone();
```

Comment:
this simply reduces the overhead of timing , right?

Reply:
Yes, which turned out to be a major bottleneck, as Instant::now is a syscall